packages/app-mobile/components/screens/LogScreen.tsx

```diff
@@ -1,37 +1,60 @@
-const React = require('react');
+import * as React from 'react';
 
-const { FlatList, View, Text, Button, StyleSheet, Platform } = require('react-native');
-const { connect } = require('react-redux');
-const { reg } = require('@joplin/lib/registry.js');
-const { ScreenHeader } = require('../ScreenHeader');
-const time = require('@joplin/lib/time').default;
+import { FlatList, View, Text, Button, StyleSheet, Platform, Share } from 'react-native';
+import { connect } from 'react-redux';
+import { reg } from '@joplin/lib/registry.js';
+import { ScreenHeader } from '../ScreenHeader';
+import time from '@joplin/lib/time';
 const { themeStyle } = require('../global-style.js');
-const Logger = require('@joplin/lib/Logger').default;
+import Logger from '@joplin/lib/Logger';
 const { BaseScreenComponent } = require('../base-screen.js');
-const { _ } = require('@joplin/lib/locale');
+import { _ } from '@joplin/lib/locale';
+import { MenuOptionType } from '../ScreenHeader';
+import { AppState } from '../../utils/types';
 
 class LogScreenComponent extends BaseScreenComponent {
-	static navigationOptions() {
+	private readonly menuOptions: MenuOptionType[];
+
+	public static navigationOptions(): any {
 		return { header: null };
 	}
 
-	constructor() {
+	public constructor() {
 		super();
 
 		this.state = {
 			logEntries: [],
 			showErrorsOnly: false,
 		};
 		this.styles_ = {};
+
+		this.menuOptions = [
+			{
+				title: _('Share'),
+				onPress: () => {
+					void this.onSharePress();
+				},
+			},
+		];
+	}
+
+	private async onSharePress() {
+		// Share all log entries included in the list. As the list only displays a subset of the log,
+		// only a subset of the log is shared.
+		const log = (this.state.logEntries as any[]).map(entry => this.formatLogEntry(entry));
```

Comment:
Yes, the entire log would indeed be better. Maybe it depends on how much data we are allowed to share. Is there any limit we should be aware of?

Reply:
At least on Android, `Share` shouldn't be used to share more than a few kilobytes of data.

On Android, React Native's `Share` [uses Intents](https://github.com/facebook/react-native/blob/eaafc260c14674bfbf7ce3a647404c70d546854f/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/modules/share/ShareModule.java#L48), which have a maximum size of around (500 KB](https://stackoverflow.com/a/37054839), but seems to vary. [The Android documentation](https://developer.android.com/guide/components/activities/parcelables-and-bundles) states that `Intent`s should be kept to no more than a few kilobytes.

If, rather than sharing the log data, we copy it directly to the clipboard, there _might_ not be a similar limit.

Reply:
A few kilobytes should be fine. If 500k is not always ok, is there a limit that is safe across devices? Then we could just use that

Reply:
[According to this StackOverflow post](https://stackoverflow.com/a/37054839), the maximum size on Android has been known to decrease across versions. Because the [(Android) documentation states "you should be careful to limit the data size to a few KB"](https://developer.android.com/guide/components/activities/parcelables-and-bundles), it might make sense to choose a maximum size like 50 KB or 100 KB.

Reply:
Perhaps an alternative would be to create a text file with the log and share that? But if it's too complicated to do so 50k would be fine, it would indeed still be better than screenshots

Reply:
It has been changed to share a text file!

The downside of this, however, is that there's no clear way to save its contents to a file or copy its contents on Android (iOS provides both options):
![screenshot: Android emulator provides no save to files option](https://github.com/laurent22/joplin/assets/46334387/677b9631-24ec-4a5a-badd-48b7ccea73c6)